src/edges.rs

```diff
@@ -36,8 +38,9 @@ pub fn canny(image: &GrayImage, low_threshold: f32, high_threshold: f32) -> Gray
     let blurred = gaussian_blur_f32(image, SIGMA);
 
     // 2. Intensity of gradients.
-    let gx = horizontal_sobel(&blurred);
-    let gy = vertical_sobel(&blurred);
+    let gradient_kernel = GradientKernel::Sobel;
+    let gx = filter3x3(&blurred, &gradient_kernel.as_horizontal_kernel::<i16>());
```

Comment:
I’m not convinced by this API - the calling code in this PR has gotten more verbose and less clear.

Reply:
In my view, the new calling code is now more clear as it uses the fundamental `filter3x3()` rather than `horizontal_sobel()` which the user wouldn't know how it was implemented. Also it allows the user to easily switch between different `GradientKernel`s without having to finding an existing method for that kernel.

I can make it more concise using some `impl Into<Kernel>` function arguments.

Reply:
Would re-adding all the `horizontal_*` `vertical_*` as aliases to the `filter()` method using the new `GradientKernel` enum be a satisfactory solution?